src/noit_metric.c

```diff
@@ -342,15 +342,10 @@ noit_metric_tagset_is_taggable_value(const char *val, size_t len)
     noit_metric_tagset_is_taggable_part(val, len, noit_metric_tagset_is_taggable_value_char);
 }
 
-static inline ssize_t
-noit_metric_tagset_encode_tag_ex(char *encoded_tag, size_t max_len,
-                                 const char *decoded_tag, size_t decoded_len,
-                                 mtev_boolean for_search,
-                                 noit_metric_encode_type_t left,
-                                 noit_metric_encode_type_t right)
-{
-  char scratch[NOIT_TAG_MAX_PAIR_LEN+1];
-  if(max_len > sizeof(scratch)) return -1;
+static inline ssize_t noit_metric_tagset_encode_tag_ex(
+    char *encoded_tag, size_t max_len, const char *decoded_tag,
+    size_t decoded_len, mtev_boolean for_search, noit_metric_encode_type_t left,
+    noit_metric_encode_type_t right) {
```

Comment:
```suggestion
    const noit_metric_encode_type_t right) {
```

Reply:
right is also mutable
`if(!for_search) left = right = '"';`